packages/typespec-vscode/package.json

```diff
@@ -108,6 +108,113 @@
+                }
+              }
+            },
+            "default": [
+              {
+                "language": "DotNet",
+                "package": "@typespec/http-client-csharp"
+              },
+              {
+                "language": "Java",
+                "package": "@typespec/http-client-java"
+              },
+              {
+                "language": "JavaScript",
+                "package": "@azure-tools/typespec-ts"
+              },
+              {
+                "language": "Python",
+                "package": "@typespec/http-client-python"
+              }
+            ],
+            "description": "Define the emitter for a language client sdk generation."
+          },
+          "typespec.server.emitter": {
+            "scope": "window",
+            "type": "array",
+            "items": {
+              "type": "object",
+              "properties": {
+                "language": {
+                  "type": "string",
+                  "enum": [
+                    "DotNet",
+                    "Java",
+                    "JavaScript",
+                    "Python",
+                    "Go"
+                  ],
+                  "description": "Define the language for the emitter."
+                },
+                "package": {
+                  "type": "string",
+                  "description": "Define the emitter package.\n\nExample (with version): @typespec/http-server-csharp@1.0.0\n\nExample (without version): @typespec/http-server-csharp"
+                }
+              }
+            },
+            "default": [
+              {
+                "language": "DotNet",
+                "package": "@typespec/http-server-csharp"
+              },
+              {
+                "language": "JavaScript",
+                "package": "@typespec/http-server-javascript"
+              }
+            ],
+            "description": "Define the emitter for server code generation."
+          },
+          "typespec.schema.emitter": {
+            "scope": "window",
+            "type": "array",
+            "items": {
+              "type": "object",
+              "properties": {
+                "language": {
+                  "type": "string",
+                  "enum": [
+                    "OpenAPI3",
+                    "ProtoBuf",
+                    "JsonSchema"
+                  ],
+                  "description": "Define the language for the emitter."
+                },
+                "package": {
+                  "type": "string",
+                  "description": "Define the emitter package.\n\nExample (with version): @typespec/openapi3@1.0.0\n\nExample (without version): @typespec/openapi3"
+                }
+              }
+            },
+            "default": [
+              {
+                "language": "OpenAPI3",
+                "package": "@typespec/openapi3"
+              }
+            ],
+            "description": "Define the emitter for a schema."
```

Comment:
can we make the settings more general? what if user want to configure other type emitters?

Reply:
@RodgeFu this is general, user can submit other type emitters. We just provide a default value openapi3